qiskit/visualization/state_visualization.py

```diff
@@ -770,13 +770,15 @@ def plot_state_qsphere(state, figsize=None, ax=None, show_state_labels=True,
 
                 # get prob and angle - prob will be shade and angle color
                 prob = np.real(np.dot(state[i], state[i].conj()))
+                if prob > 1:  # See https://github.com/Qiskit/qiskit-terra/issues/4666
+                    prob = 1
                 colorstate = phase_to_rgb(state[i])
 
                 alfa = 1
                 if yvalue >= 0.1:
                     alfa = 1.0 - yvalue
 
-                if prob > 0 and show_state_labels:
+                if not np.isclose(prob, 0) and show_state_labels:
```

Comment:
The only question I have here is can prob ever be < 0 here? If it is and is outside the default tolerances on `isclose` this if will be triggered now but wouldn't be before.

Reply:
It should not by construction be less than 0, by construction, if I understand the code correctly. What it can happen is that is negative because unrounding (like the `prob < 1` situation) and `isclose` should capture that.